metapool/literals.py

```diff
@@ -0,0 +1,112 @@
+# I am aware that "having a global constants class" is largely regarded as
+# an antipattern in Python because it increases dependence between the
+# modules using it and keeps pieces tightly coupled that should be loosely
+# coupled.  However, the current codebase has some literal strings repeated
+# *throughout* the code, and I need them all in one place while we get them
+# replaced with constants so that we can (a) prevent re-definition of the same
+# constant string in multiple places and (b) get a clear view of which of
+# them are really cross-domain and which really do belong to a specific
+# domain (where they can then be moved in the future).  Constants that I
+# already *know* to be domain-specific are not included here.
+
+# Note:
+# A number of these string literals are also used in amplicon-related code
+# (e.g., prep.py, scripts/seqpro.py, seqpro_mf.py, etc.).  However, I don't
+# know if they are the same because they come from the same source, or if it is
+# just coincidental.  For now, I have NOT changed the amplicon-related
+# instances to use these constants.
+
+import re
+
+# These are the "standard"-format snake-case column names;
+# if writing something NEW, use these
+SAMPLE_NAME_KEY = "sample_name"
+QIITA_ID_KEY = "qiita_id"
+SAMPLE_PROJECT_KEY = "sample_project"
+ORIG_NAME_KEY = "orig_name"
+EXPT_DESIGN_DESC_KEY = 'experiment_design_description'
+CONTAINS_REPLICATES_KEY = "contains_replicates"
+PROJECT_SHORT_NAME_KEY = "short_project_name"
+PROJECT_FULL_NAME_KEY = "full_project_name"
+# NB: don't use this as the name for a column of sample identifiers.  It is
+# the key for a list of sample info *objects* (e.g., dictionaries)
+SAMPLES_KEY = "samples"
+
+# Plate map (PM) column names
+PM_SAMPLE_KEY = "Sample"
+PM_PROJECT_PLATE_KEY = "Project Plate"
+PM_PROJECT_NAME_KEY = "Project Name"
+PM_PROJECT_ABBREV_KEY = 'Project Abbreviation'
+PM_COMPRESSED_PLATE_NAME_KEY = "Compressed Plate Name"
+PM_BLANK_KEY = "Blank"
+PLATE_NAME_DELIMITER = "_"
+
+# Currently used to describe sample context (only)
+SAMPLE_TYPE_KEY = "sample_type"
+PRIMARY_STUDY_KEY = "primary_qiita_study"
+SECONDARY_STUDIES_KEY = "secondary_qiita_studies"
+
+
+def get_plate_name_delimiter():
+    return PLATE_NAME_DELIMITER
```

Comment:
Out of curiosity, why have a function for this?

Reply:
That is true, getters() are IMHO less pythonic than the @property decorator. I tend to just directly use the variable since all members are public in Python but I believe the clean way is to use a property decorator.

Reply:
I believe that an  @property decorator cannot be used here because there is no class for the property to be attached to.  I am willing to take this out.

Reply:
> I believe that an @Property decorator cannot be used here because there is no class for the property to be attached to. I am willing to take this out.

Good point 😆  Since it's not in a class using the literal directly seems cleaner.